### PR TITLE
Add coroutines support

### LINKED ISSUE #132: Explore Kotlin coroutine
* Explore how Kotlin coroutine works
* See how to integrate with `FirestoreDatabase` or `Table` if applicable

### PR DESCRIPTION
**Pull request by Plouc314**
* Replace all methods with `onCompletion` callback by `suspend` method, in `FirestoreDatabase`, `Table` and children
* Adapt `CalendarViewModel`, `FlightsViewModel` to properly call the `suspend`methods
* Refactored existing database-related tests

Close #132

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/skysync/database/tables/FlightTableUnitTest.kt [hunk 1]
```diff
@@ -17,134 +17,49 @@ import org.junit.runner.RunWith
 @RunWith(AndroidJUnit4::class)
 class FlightTableUnitTest {
   private val db = FirestoreDatabase(useEmulator = true)
-  private val databaseSetup = DatabaseSetup()
+  private val dbs = DatabaseSetup()
   private val flightTable = FlightTable(db)
```
Comment (Baribal02) [inline app/src/androidTest/java/ch/epfl/skysync/database/tables/FlightTableUnitTest.kt:21]: I prefer the use of databaseSetup as it makes it clearer.

File: app/src/main/java/ch/epfl/skysync/database/Table.kt [hunk 2]
```diff
@@ -16,48 +16,65 @@ abstract class Table<M, S : Schema<M>>(
...
   /**
    * Get an item by ID
    *
    * @param id The id of the item
-   * @param onCompletion Callback called on completion of the operation
    * @param onError Callback called when an error occurs
    */
-  open fun get(id: String, onCompletion: (M?) -> Unit, onError: (Exception) -> Unit) {
-    db.getItem(path, id, clazz, { onCompletion(it?.toModel()) }, onError)
+  open suspend fun get(id: String, onError: ((Exception) -> Unit)? = null): M? {
```
Comment (Baribal02) [inline app/src/main/java/ch/epfl/skysync/database/Table.kt:47]: What does the open keyword ?

**Review by Baribal02 — APPROVED** [review]
Overall looks good to me

File: app/src/main/java/ch/epfl/skysync/database/Table.kt (hunk 2 above)
Comment (Plouc314) [reply app/src/main/java/ch/epfl/skysync/database/Table.kt:47]: It used to declare that a function can be overridden, as methods are by default final in Kotlin.

**Pull request merged**